dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,323 @@
+import math
+import numpy as np
+import toolz
+from sklearn.base import clone
+from sklearn.model_selection import ParameterSampler
+from sklearn.utils import check_random_state
+
+from tornado import gen
+from toolz import groupby
+from dask.distributed import as_completed, default_client, futures_of
+from dask_searchcv.model_selection import DaskBaseSearchCV
+import dask.array as da
+from ..datasets import make_classification
+from ..wrappers import Incremental
+from ._split import train_test_split
+from time import time
+
+
+def _get_nr(R, eta=3):
+    s_max = math.floor(math.log(R, eta))
+    B = (s_max + 1) * R
+
+    brackets = list(reversed(range(int(s_max + 1))))
+    N = [math.ceil(B / R * eta**s / (s + 1)) for s in brackets]
+    R = [int(R * eta**-s) for s in brackets]
+    return list(map(int, N)), list(map(int, R)), brackets
+
+
+def _partial_fit(model_meta_future, X, y, meta=None, fit_params={},
+                 dry_run=False, **kwargs):
+    start = time()
+    assert isinstance(X, np.ndarray)
+    assert y is None or isinstance(y, np.ndarray)
+    model, m = model_meta_future
+    if meta is None:
+        meta = m
+    while meta['iterations'] < meta['partial_fit_calls']:
+        if not dry_run:
+            model.partial_fit(X, y, **fit_params)
+        meta["iterations"] += 1
+    meta['mean_fit_time'] += time() - start
+    return model, meta
+
+
+def _score(model_and_meta, x, y, dry_run=False):
```

Comment:
Another TODO: The user provided `scoring` to `HyperBandCV` should be plumbed all the way down here.

Scikit-learn uses a `_passthrough_scorer`: https://github.com/scikit-learn/scikit-learn/blob/3b5abf76597ce6aff76192869f92647c1b5259e7/sklearn/metrics/scorer.py#L226-L228

We could copy that, or ask if it can be made public.

Reply:
Shouldn't we use the result of `check_scoring`? That's what sklearn eventually calls in their BaseSearchCV.

Reply:
I think this is resolved.